its/core-it-support/core-it-plugins/maven-it-plugin-active-collection/src/main/java/org/apache/maven/plugin/coreit/CheckThreadSafetyMojo.java

```diff
@@ -95,7 +95,11 @@ public void run() {
                     getLog().info("[MAVEN-CORE-IT-LOG] Thread " + this + " uses " + tccl);
                     Thread.currentThread().setContextClassLoader(tccl);
                     while (go.isEmpty()) {
-                        // wait for start
```

Comment:
no need for debounce?

potentially `while(true)` cloud max out CPU.

Reply:
I have not checked the context. But maybe `java.util.concurrent.CountDownLatch` could be used instead of that loop.

Reply:
> CountDownLatch

thanks for ideation.

Reply:
https://github.com/apache/maven/pull/2381